contracts/IbbtcVaultZap.sol

```diff
@@ -146,8 +146,76 @@ contract IbbtcVaultZap is PausableUpgradeable {
             );
     }
 
+    function _vaultShares(uint256 _amount)
+        internal
+        view
+        returns (uint256 shares)
+    {
+        uint256 totalSupply = IBBTC_VAULT.totalSupply();
+        shares = _amount;
+        if (totalSupply != 0) {
+            shares = (_amount.mul(totalSupply)).div(IBBTC_VAULT.balance());
+        }
+    }
+
+    function _calcIbbtcMint(uint256[2] memory _amounts)
+        internal
+        view
+        returns (uint256 ibbtc)
+    {
+        uint256 lp;
+        uint256 amount;
+        uint256 bBTC;
+        uint256 fee;
+        uint256 sett;
+        uint256[2] memory depositAmounts;
+        depositAmounts[0] = _amounts[0];
+        for (uint256 i; i < 2; i++) {
+            if (_amounts[i] != 0) {
+                // for renBTC [amount, 0] and wBTC [0, amount]
+                lp = CURVE_REN_POOL.calc_token_amount(depositAmounts, true);
+                sett = lp.mul(1e18).div(RENCRV_VAULT.getPricePerFullShare());
+                (bBTC, fee) = SETT_PEAK.calcMint(0, sett);
+                ibbtc += bBTC;
+            }
+            depositAmounts[0] = 0;
+            depositAmounts[1] = _amounts[1];
+        }
+    }
```

Comment:
Why not call `calc_token_amount` just once with `_amounts`?

Reply:
makes sense, done